src/VSCodeExtension/src/extension.ts

```diff
@@ -200,6 +200,94 @@ export function activate(context: vscode.ExtensionContext) {
     ).then((dotNetSdk) => {
         console.log(`[qsharp-lsp] Found the .NET Core SDK at ${dotNetSdk.path}.`);
 
+        // Register commands that use the .NET Core SDK.
+        context.subscriptions.push(
+            vscode.commands.registerCommand(
+                "quantum.installTemplates",
+                () => {
+                    let packageVersion = 
+                        packageInfo === undefined
+                        ? ""
+                        : `::${packageInfo.version}`;
+                    cp.spawn(
+                        dotNetSdk.path,
+                        ["new", "--install", `Microsoft.Quantum.ProjectTemplates${packageVersion}`]
+                    ).on(
+                        'exit', (code, signal) => {
+                            if (code === 0) {
+                                vscode.window.showInformationMessage(
+                                    "Project templates installed successfully."
+                                );
+                            } else {
+                                vscode.window.showErrorMessage(
+                                    `.NET Core SDK exited with code ${code} when installing project templates.`
+                                );
+                            }
+                        }
+                    );
+                }
+            )
+        );
+        
+        context.subscriptions.push(
+            vscode.commands.registerCommand(
+                "quantum.newProject",
+                () => {
+                    // Where should we make the new project?
+                    vscode.window.showSaveDialog({
+                        saveLabel: "Create Project"
+                    }).then(
+                        (uri) => {
+                            if (uri !== undefined) {
+                                if (uri.scheme !== "file") {
+                                    vscode.window.showErrorMessage(
+                                        "New projects must be saved to the filesystem."
+                                    );
+
+                                }
+                                // TODO: show quick pick between console, library, and tests.
+                                cp.spawn(
+                                    dotNetSdk.path,
+                                    ["new", "console", "-lang", "Q#", "-o", uri.fsPath]
```

Comment:
Looks like making the distinction for different kinds of project seems fairly straighforward - do you want to add that or shall I?

Reply:
Just took a stab at it, but I'd appreciate a test to make sure I got the UX right.